apps/andi/lib/andi_web/controllers/api/ingestion_controller.ex

```diff
@@ -144,11 +144,27 @@ defmodule AndiWeb.API.IngestionController do
   end
 
   defp add_uuid(message) do
-    uuid = UUID.uuid4()
-
-    Map.merge(message, %{"id" => uuid}, fn _k, v1, _v2 -> v1 end)
+    if Map.has_key?(message, "id") do
+      case IngestionStore.get(Map.get(message, "id")) do
```

Comment:
Is this ingestion store the right way to do this? Super worried that a mistake will just stop updates from going through to ingestions

Reply:
It should be Ingestions.get(id)

IngestionStore is essentially the cache across all services. It represents what the system knows about the ingestion as a whole. Generally the result of messages being processed.

Ingestions is a wrapper around the postgres database backing Andi. 

So Andi likely will always use Ingestions, then only after it's done all of it's work, finally send a ingestion_update message, which is processed and the IngestionStore is updated.

Simple, right?

Reply:
So I copied this over from the GET endpoint, figuring consistency was the way to go. Do you think we should change that endpoint to also be hitting postgres?

Reply:
Hm, I see why'd you reference that yeah.

Since IngestionStore is returning from the ViewState, the response will be a SmartCity struct cast to JSON. I believe if we instead used "Ingestions.get", we'd get back an AndiSchema.Ingestion response.

Because DatasetController and OrganizationController both use the ViewState Stores (Ex: IngestionsStore like you have here), my vote would be to stick with using IngestionStore, as you have it now, for consistency.

If later we want to have a conversation about switching to returning to Andi Schema responses across the API, I'd be open to that depneding on folks use case other than our internal testing, publish / draft status etc, but that convo could be down the road IMO.